File: apple/Sources/ReactNativeWatchOS/Modifiers/LabelsHidden.swift
import SwiftUI

/// SwiftUI `.labelsHidden()`. Hides the labels of controls (e.g. a `Toggle`,
/// `Picker`, or `Slider` title) while preserving them for accessibility.
///
/// Available on watchOS 6+, so no availability gate is needed.
enum RNWLabelsHiddenModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("labelsHidden") { view, _, _ in
            AnyView(view.labelsHidden())
        }
    }
}

File: src/modifiers/labelsHidden.ts
import { createModifier } from './createModifier';

/// SwiftUI `.labelsHidden()`. Hides the labels of controls (e.g. the title of a
/// `Toggle`, `Picker`, or `Slider`) while keeping them for accessibility.
export function labelsHidden() {
  return createModifier('labelsHidden');
}
